sklearn/model_selection/_validation.py

```diff
@@ -639,9 +639,20 @@ def cross_val_predict(estimator, X, y=None, groups=None, cv=None, n_jobs=1,
 
     cv = check_cv(cv, y, classifier=is_classifier(estimator))
 
-    if method in ['decision_function', 'predict_proba', 'predict_log_proba']:
-        le = LabelEncoder()
-        y = le.fit_transform(y)
+    # If classification methods produce multiple columns of output,
+    # we need to manually encode classes to ensure consistent column ordering.
+    encode = method in ['decision_function', 'predict_proba',
+                        'predict_log_proba']
+    if encode:
```

Comment:
These changes conflict with #9593, right?

Reply:
If I read #9593 correctly, the change in https://github.com/scikit-learn/scikit-learn/pull/8773/files#diff-60033c11a662f460e1567effd5faa6f0R784 (the first line of `_enforce_prediction_order`) is the same as the code change in that PR.